Restructuration du README et déplacement de certains éléments dans CO… (#1024)
python/python-docs-fr: CONTRIBUTION.rst

> Guide de contribution à la documention via GitHub.

Reviewer: Ce fichier est référencé dans le readme ? (Je suis en train de lire hein, c'est des questions bêtes peut etre)

Author: J'ai mis un lien (URL) dans le README qui pointe vers CONTRIBUTION

===================================================

Prérequis :
------------

- Un compte `Github <https://github.com/join>`_.
- ``git`` `installé <https://help.github.com/articles/set-up-git/>`_ (pour Windows, voir
  https://gitforwindows.org/).
- Un éditeur de fichier ``.po`` (comme `poedit <https://poedit.net/>`_).

Instructions :
--------------

Pour commencer vous aurez besoin de *forker* le dépôt des sources `python-docs-fr
<https://github.com/python/python-docs-fr>`_ en cliquant sur son bouton
``Fork``. Ceci crée une copie du projet sur votre compte Github : un endroit
où vous avez le droit de faire des modifications.

Étape par étape :

.. code-block:: bash

    # Clonez votre fork Github avec `git` en utilisant ssh :
    git clone git@github.com/VOTRE_NOM_DE_COMPTE_GITHUB/python-docs-fr.git

    # *OU* HTTPS :

>     git clone https://github.com/VOTRE_NOM_DE_COMPTE/python-docs-fr.git

Reviewer 2: J'aurais mis VOTRE_NOM_DE_COMPTE_GITHUB pour être homogène avec juste au-dessus.

Reviewer: ```suggestion
    git clone https://github.com/VOTRE_NOM_DE_COMPTE_GITHUB/python-docs-fr.git
```

    # Allez dans le répertoire cloné :
    cd python-docs-fr/

>     # Ajoutez le dépôt public (nommé upstream)

Reviewer 2: ```suggestion
    # Ajoutez le dépôt officiel (nommé upstream)
```

Reviewer: ```suggestion
    # Ajoutez le dépôt officiel en upstream
```
?

Author: Upstream est le nom de la branche donc je pense que c'est plus clair de laisser "(nommé upstream)"

    # Ceci permet à *git* de savoir quoi/où est *upstream*
    git remote add upstream https://github.com/python/python-docs-fr.git

Ensuite, vous devez trouver un fichier sur lequel travailler

> (pour vous aiguillez vous pouvez vous rendre à `Que traduire ?`_).

Reviewer: Ici tu fait référence à potodo ?

Author: Yep, quand on suit le lien on tombe sur les indications et finalement sur potodo

Une fois que vous avez choisi un fichier sur lequel travailler, veuillez
ouvrir un `ticket sur Github <https://github.com/python/python-docs-fr/issues>`_ dans

> le format ``Je travaille sur RÉPERTOIRE/FICHIER.po``.

Reviewer: ```suggestion
le format ``Je travaille sur DOSSIER/FICHIER.po``.
```
? Pas très important 😄

> Ceci permet à `potodo`_ de détecter via l'API Github les fichiers ``.po`` réservés

Reviewer: ici, c'est un lien le 
```
`potodo`_
```
?

Reviewer 3: Peut-être faut-il mentionner le template prédéfini ? https://github.com/python/python-docs-fr/pull/1029.

Author: J'ai modifié la formulation pour que le format du titre soit plus explicite.

dans les tickets et les *pull requests*.

N'hésitez pas non plus à vous équiper de quelques outils pour vous aider dans
votre traduction (voir `Outils utiles pour la traduction`_)

Vous êtes maintenant prêt à commencer une session de travail. Chaque

> fois que vous commencerez un nouveau fichier, commencez ainsi :

Reviewer 2: Il y a trois fois le mot « commencer », on pourrait en remplacer par « débuter », « démarrer » ou autres.

Reviewer: Il faudrait aussi rajouter que il faut faire l'étape `potodo` et `issue` /!\

Author: @Seluj78 Je n'ai pas compris ta remarque :/

Reviewer: Je propose, dans cette section ou on explique la démarche a suivre a chaques fois que on commence un nouveau fichier, de rajouter une partie sur potodo et de creer une issue

.. code-block:: bash

> 

Reviewer: `git checkout 3.8` d'abord non ? ou pas nécéssaire car on base les branches sur 3.8 ligne 63 ?

Reviewer 4: pas nécessaire car il ne se base pas sur `HEAD` (ni sur `3.8`) mais sur `upstream/3.8`.

C'est un bon moyen de faire une nouvelle branche propre : Même si l'utilisateur à posé des commits sur sa 3.8 par erreur, le `fetch` va réussir (ce n'est qu'un téléchargement, il ne touche pas aux branches locales, encore moins au *working tree*), et la création de branche va réussir aussi.

Alors qu'avec la méthode "plus classique" de `git checkout 3.8 ; git pull` on a le risque que le `git pull` ne soit pas « fast forward », qu'il crée un *merge* commit, voir qu'il trouve des conflits, et que les gens basent leur branches dessus sans s'en rendre compte.

Author: @JulienPalard merci pour les explications !
Je peux marquer comme résolu ?

    # Pour travailler, nous aurons besoin d'une branche, basée sur une version à jour
    # (fraîchement récupérée) de la branche upstream/3.8. On met donc à jour notre
    # version locale.
    git fetch upstream

    # On créé ensuite une branche. Il est pratique de nommer la branche en fonction du
    # fichier sur lequel on travaille. Par exemple, si vous travaillez sur
    # « library/venv.po », vous pouvez nommer votre branche « library-venv ».
    # Cette nouvelle branche nommée « library-sys » est basée sur « upstream/3.8 ».

>     git checkout -b library-sys upstream/3.8

Reviewer 2: La phrase précédente par de librairy/venv.po, ça serait plus cohérent de faire tout l'exemple avec.

    # Vous pouvez maintenant travailler sur le fichier (typiquement, en utilisant poedit).
    # N'oubliez pas de configurer poedit pour passer à la ligne à 79 caractères.
    # (Édition -> Préférences -> Avancé -> Passer à la ligne à : 79)

>     # Ici, remplacez « library/sys.po » par le fichier que vous avez choisi précédemment.

Reviewer: Preciser aussi qq part que on peux preciser dans poedit non prenom email?

Reviewer 4: Normalement c'est la configuration par défaut 79, si je ne m'abuse (je n'utilise pas poedit).

Reviewer 2: Dans le poedit que j'ai (Linux Mint 19.2), c'est effectivement déjà pré-configuré à 79. Après je ne sais pas dire si c'est partout le cas.

Author: Version 2.2.4 de Poedit le nombre de lignes est effectivement pré-configuré, mais il faut saisir le nom/email effectivement. Je rajoute.

>     poedit library/sys.po

Reviewer: Toutes les distributions n'ont pas la commande poedit !

Reviewer 4: Donc : 
```suggestion
    poedit library/sys.po  # ou lancez simplement poedit puis « Fichier » → « Ouvrir »
```

(à vérifier que c'est bien fichier et ouvrir :P)

Reviewer 2: > à vérifier que c'est bien fichier et ouvrir :P

Oui, c'est ça ! :wink: Je viens de vérifier.

>     # Si vous n'utilisez pas poedit, vous pouvez utiliser `powrap` (voir la section *outils*)

Reviewer: Remplacer ca par un lien ? Peut etre du coup reduire la section code block en plein de petit code blocks ?

Author: Impossible dans un commentaire de code.
Il faudrait effectivement découper tous le tuto en pleins de petits code blocks, à discuter.

Reviewer: Peut être une fois mergé

    # qui va reformater correctement le fichier que avez vous avez modifié.
    # Exécutez `powrap -m` (reformater tous les fichiers modifiés)
    # ou `powrap library/sys.po` (un fichier en particulier) :
    powrap -m

    # Pour l'orthographe, une liste blanche de certains termes techniques ou
    # de noms propres, comme « Guido », « C99 » ou « sérialisable », est
    # stockée dans le fichier « dict » à la racine du projet.  Vous pouvez

>     # bien sûr y ajouter une entrée si nécessaire.

Reviewer: ```suggestion
    # bien sûr y ajouter une entrée si nécessaire. Cette commande va lancer les vérifications nécéssaires
```

    make verifs

    # C'est le moment de git add et git commit

>     # C'est l'occasion de se relire avec l'option -p ou -u.

Reviewer 2: L'option -p ou -u de quoi ?

Reviewer: de `git add` mais faut en effet expliquer. Je pense que il faut diviser ce code block en plus petit

Reviewer 4: `git add -p` ça doit venir de moi, c'est ce que j'utilise peu importe le projet, ça permet d'ajouter petit patch par petit patch, et donc en passant ça permet de se relire.

`git add -u` c'est pas mal aussi : ça prend tout ce qui à été modifié.

Author: `git add -p` c'est un peu le `progrep` de git ;)
Est-ce que l'on découpe cette partie en petits blocks comme le propose @Seluj78 pour gagner en clarté?

Reviewer 4: La découpe est probablement une bonne idée, mais pas dans la même PR, sinon on aura toujours des bonnes idées, et on ne mergera jamais. Y'a déjà du bon travail, je veux merger ça, c'est utile.

    git add library/sys.po

    git commit -m "Traduction de library/sys.po"  # Ou un autre message plus inspiré :)

    # Poussez ensuite vos modifications sur votre fork Github.
    # Le -u n'est utile qu'une fois pour que votre client git se souvienne que cette
    # branche est liée à votre fork Github (et donc qu'un futur `git pull` sache quoi

>     # tirer)

Reviewer: Et qu'il suffira d'un `git push` plus tard

    git push -u origin

    # La commande précédente vous affichera un lien pour ouvrir une pull request sur
    # Github. Si vous l'avez manqué, allez simplement sur https://github.com/python/python-docs-fr/pulls
    # et un joli bouton « Compare & pull request » devrait apparaître au bout de
    # quelques secondes vous indiquant que vous pouvez demander une pull request.

>     # À partir de là, quelqu'un passera en revue vos modifications, et fera desvous voudrez

Reviewer 2: ```suggestion
    # À partir de là, quelqu'un passera en revue vos modifications, et vous fera des
```

    # suggestions/corrections. Pour les prendre en compte, retournez sur votre branche
    # contenant du fichier concerné (au cas où vous auriez commencé quelque chose d'autre
    # sur une autre branche) :
    git checkout library/sys
    git pull  # pour rapatrier les modifications que vous auriez accepté
              # sur l'interface web.

    # Réglez les problèmes, puis commitez à nouveau :

>     git commit -a -m "sys : corrections".

Reviewer 2: ```suggestion
    git commit -a -m "prise en compte des remarques"
```

À voir pour le message. Par contre il y a un point en trop à la fin de la phrase.

Reviewer 4: Après de toute façons on `squash and merge` les PRs, donc ce message va disparaître.

Author: @JulienPalard Je maintiens la correction de @vpoulailleau en tant que première version.

    git push

Vous avez peut-être remarqué que cela ressemble à un triangle, avec un
segment manquant :

- Vous récupérez depuis *upstream* (le dépôt commun public sur Github)
- Vous poussez sur *origin* (votre clone sur Github)

C'est le travail de quelqu'un d'autre d'ajouter le dernier segment,
de votre *origin* au *upstream* public, pour « boucler la boucle ». C'est le
rôle des personnes qui *fusionnent* les *pull requests* après les avoir relues.

Vous avez peut-être aussi remarqué que vous n'avez jamais commité sur une
branche de version (``3.7``, ``3.8``, etc.), seulement récupéré les
modifications à partir d'elles.

Toutes les traductions sont faites sur la dernière version.
Nous ne traduisons jamais sur une version plus ancienne. Par exemple,
si la dernière version de python est Python 3.8, nous ne voulons pas
traduire directement sur la version python 3.5.
Si nécessaire, les traductions seraient rétroportées sur les versions
les plus anciennes par l'`équipe de documentation
<https://www.python.org/dev/peps/pep-8015/#documentation-team>`_.

Que traduire ?
~~~~~~~~~~~~~~

Vous pouvez utiliser `potodo`_, un outil fait pour trouver des fichiers ``po``
à traduire.

> Une fois installé, utiliser la commande ``potodo`` dans votre clone local.

Reviewer 2: Renvoyer vers la section « outils » ?

Author: `potodo`_ est un lien qui renvoit déjà vers la section outil.
Je ne pas sûr de comprendre ta remarque du coup :/

car c'est une partie très technique.**

Vous pouvez commencer par des tâches faciles comme réviser les entrées
*fuzzy* pour aider à garder la documentation à jour (trouvez les entrées
*fuzzy* l'aide de `make fuzzy`).

Vous pouvez également relire des entrées déjà traduites pour vous faire une
idée,et passer ensuite à la traduction de celles qui ne sont pas traduites.
Vous pouvez les trouver à l'aide de `make todo`...

Si vous traduisez un titre de lien, veuillez traduire le lien aussi.
(surtout si c'est un lien Wikipédia et que l'article a une traduction).
Si aucune traduction de la cible n'existe, ne traduisez pas le titre.

Dans les fichiers, ne traduisez pas le contenu des balises telles que
``:ref :...`` et ``:term :...``.

Si vous devez utiliser des mots anglais, mettez les en *italique*
(entourés par des astérisques).

Pour les caractères spéciaux, référez vous à la section
`Les caractères spéciaux`_

Utilisation du futur
~~~~~~~~~~~~~~~~~~~~

Dans la description du comportement de Python (au sens large, c'est-à-dire
l'interpréteur lui-même mais aussi toutes les bibliothèques), la version
originale utilise souvent le futur : « if you do this, il will produce that … ».

> En français, l'utlisation du présent convient tout à fait et le présent est

Reviewer 2: ```suggestion
En français, l'utilisation du présent convient tout à fait et le présent est
```

l'outil graphique de configuration de votre clavier ou via
``dpkg-reconfigure keyboard-configuration``
(pour `Ubuntu <https://help.ubuntu.com/community/ComposeKey>`_ ou Debian
et distributions assimilées).

À minima, vous pouvez configurer votre fichier '~/.Xmodmap' pour
ajouter l'équivalent de :

.. code-block:: shell

    # key Compose
    keycode 115 = Multi_key

Utilisez ``xev`` pour connaitre la bonne correspondance de la touche que vous
voulez assigner !

Ensuite, dans votre fichier '~/.xsession', ajoutez :

.. code-block:: shell

    # Gestion des touches clavier
    xmodmap $HOME/.Xmodmap

Sous X, avec un bureau graphique, tel que Gnome, ou Xfce, il faut aller
modifier dans les paramètres > clavier > Disposition : puis
'Touche composée'. Pour finir, redémarrez votre session.

=> Sous Windows, vous
pouvez utiliser `wincompose <https://github.com/SamHocevar/wincompose>`_.

Le cas de « --- », « -- »,  « ... » :
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

La version anglaise utilise une chose nommée
`smartquotes <http://docutils.sourceforge.net/docs/user/smartquotes.html>`_,
qui fonctionne en anglais, mais cause des problèmes dans d'autres langues.

> Nous l'avons donc désactivée.

Reviewer 2: ```suggestion
Nous l'avons donc désactivée dans la version française.
```

Les *smartquotes* sont normalement responsables de la transformation de
``--`` en *en-dash* (``—``), de ``---`` en *em-dash* (``—``), et de
``...`` en *ellipses* ``…``.

=> Si vous voyez :
| « -- » ou « --- » : faites « Compose » + « - » ×3

> | « ... » : faites « Compose » + « - » ×3

Reviewer 2: ```suggestion
| « ... » : faites « Compose » + « . » ×3
```

Le cas de « "…" » :
~~~~~~~~~~~~~~~~~~~

Les guillemets français ``«`` et ``»`` ne sont pas identiques aux
guillemets anglais ``"``. Cependant, Python utilise les guillemets
anglais comme délimiteurs de chaîne de caractères. Il convient donc de
traduire les guillemets mais pas les délimiteurs de chaîne.

=> Si vous voyez :

> | « "…" » : faites « Compose » + « << » ×2 ou « >> » ×2

Reviewer 2: ```suggestion
| « "…" » : faites « Compose » + « < » ×2 ou « > » ×2
```
Ou alors il faut supprimer les x2

Reviewer 4: Je suis pour supprimer les ×2, pourquoi pas essayer :

    faites : :kbd:`Compose-<-<`

Reviewer 2: Je suis pour l'usage de `:kbd:` qui est supporté par GitHub (https://github.com/github/markup/issues/497).

Par contre, à ce que j'en comprends en lisant différentes docs, c'est que

```
:kbd:`Compose-<-<`
```

Signifie appuyer sur compose, sur < et sur < *simultanément*

Pour un aspect séquentiel, j'ai compris qu'il fallait utiliser : 

```
:kbd:`Compose < <`
```

Vos avis ?

Reviewer 4: Bonne remarque ! Selon la [doc de sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-kbd)

> may depend on platform- or application-specific conventions

ce qui ne nous aide pas beaucoup. Ce que tu indique me rappelle la syntaxe emacs, probablement utilisé par d'autres, où `C-u 4 C-d` indique bien `C-u` ensemble puis `4` puis `C-d`.

Je vote pour

    :kbd:`Compose < <`

Reviewer 2: > Ce que tu indique me rappelle la syntaxe emacs

Effectivement ! Surtout quand on parle de raccourci clavier compliqué :stuck_out_tongue:

Reviewer 2: Et au passage, cette remarque s'applique à tous les raccourcis clavier du document !

Author: Done.

Le cas de « :: »
~~~~~~~~~~~~~~~~

| Du point de vue du langage *reStructuredText* (ou *rst*) utilisé dans la
  documentation nous voyons soit « bla bla:: », soit « bla bla. :: ».
| ``::`` collé à la fin d'un mot signifie « affiche ``:`` et introduit un bloc de code »,
  mais un ``::`` après une espace signifie « introduit juste un bloc de code ».

En français, nous mettons une espace insécable devant nos deux-points, comme :
« Et voilà : ».

=> Traduisez ``mot deux-points deux-points`` par
``mot espace-insécable deux-points deux-points``.

Pour saisir une espace insécable faites : « Compose » + « espace » ×2

Outils utiles pour la traduction
--------------------------------

Potodo
~~~~~~

| Permet de d'identifier les parties de la documention qu'il reste à traduire.
| Installez-le à l'aide de *pip* (``pip install potodo``) dans un environnement
  ``python3.6`` ou plus.
| `Lien vers le repository <https://github.com/seluj78/potodo>`__

Pogrep
~~~~~~

> | Permet de rechercher dans la documention des termes. Utiles si on a un doute

Reviewer 2: ```suggestion
| Permet de rechercher dans la documentation des termes. Utile si on a un doute
```

  sur comment traduire un terme ou chercher la traduction d'un terme dans
  d'autres fichiers.
| Installez-le à l'aide de *pip* (``pip install pogrep``) dans un
  environnement.
| `Lien vers le repository <https://github.com/JulienPalard/pogrep>`__

> Padpo (beta)

Reviewer: @vpoulailleau 🎉

Reviewer 2: La classe ! :wink:

~~~~~~~~~~~~~~~

| Analyseur de code qui vérifie la grammaire et l'orthographe et la syntaxe
  du fichier .po.
| Installez-le à l'aide de *pip* (``pip install padpo``) dans un environnement
  ``python3.7`` ou plus.
| `Lien vers le repository <https://github.com/vpoulailleau/padpo>`__

Powrap
~~~~~~

| Formateur de fichier .po.
| Installez-le à l'aide de *pip* (``pip install powrap``) dans un
  environnement.
| `Lien vers le repository <https://github.com/JulienPalard/powrap>`__

Ressources de traduction
-------------------------

> - le canal IRC `#python-fr <http://irc.lc/freenode/python-fr>`_ sur freenode ;

Reviewer: aussi `python-docs-fr`

Author: C'est un autre canal IRC ?

Reviewer 4: On peut conseriller le chan anglophone `#python-doc`.

- les `glossaires et dictionnaires de traduc.org
  <https://traduc.org/Glossaires_et_dictionnaires>`_, en particulier le
  `grand dictionnaire terminologique <http://gdt.oqlf.gouv.qc.ca/>`_
  de l'Office québécois de la langue française ;
- le `glossaire Python <https://docs.python.org/fr/3/glossary.html>`_, car
  il est déjà traduit ;
- le `guide stylistique pour le français de localisation des produits Sun
  <https://web.archive.org/web/20160821182818/http://frenchmozilla.org/FTP/TEMP/guide_stylistique_December05.pdf>`_ donne
  beaucoup de conseils pour éviter une traduction trop mot à mot ;
- `deepl.com/translator <https://www.deepl.com>`_ ;
- `Petites leçons de typographie <https://jacques-andre.fr/faqtypo/lessons.pdf>`_,
  résumé succint de typographie, utile pour apprendre le bon usage des
  majuscules, des espaces, etc.

Glossaire
---------

Afin d'assurer la cohérence de nos traductions, voici quelques propositions et
rappels pour les termes fréquents à traduire, n'hésitez pas à ouvrir un ticket
si vous n'êtes pas d'accord.

Pour trouver facilement comment un terme est déjà traduit dans notre documentation,
vous pouvez utiliser

> `find_in_po.py <https://gist.github.com/JulienPalard/c430ac23446da2081060ab17bf006ac1>`_.

Reviewer: Soon, y'aura une commande comme powrap etc... cc @JulienPalard

Author: C'est pogrep ? Je n'ai pas relu cette section là et ai fais un bête copy/paste.
Je peux remplacer cette ligne par la mention de pogrep, qui me semble plus pertinente, pour un meilleur résultat.

Reviewer 4: En effet c'est pogrep maintenant : pypi.org/p/pogrep

Author: @JulienPalard Ok noté

library                    bibliothèque
list comprehension         liste en compréhension (liste en intension est
                           valide, mais nous ne l'utilisons pas)
little-endian, big-endian  `petit-boutiste, gros-boutiste
                           <https://fr.wikipedia.org/wiki/Endianness>`_
mixin type                 type de mélange
mutable                    muable
namespace                  espace de nommage
                           (sauf pour le XML où c'est espace de noms)
parameter                  paramètre
pickle (v.)                sérialiser
prompt                     invite
raise                      lever
regular expression         expression rationnelle, expression régulière
return                     renvoie, donne (on évite « retourne » qui
                           pourrait porter à confusion).
setter                     mutateur
simple quote               guillemet simple
socket                     connecteur ou interface de connexion
statement                  instruction
subprocess                 sous-processus
thread                     fil d'exécution
traceback                  trace d'appels, trace de pile

> underscore                 tiret bas, *underscore*

Reviewer: ```suggestion
underscore                 tiret bas, *underscore*
whitespace                 caractère d'espacement
```
cc @christopheNan

Guide de contribution à la documention via GitHub.
===================================================

> Prérequis :

Reviewer 4: C'est un titre, je doute que les deux points soit intéressants.

------------

- Un compte `Github <https://github.com/join>`_.

> - ``git`` `installé <https://help.github.com/articles/set-up-git/>`_ (pour Windows, voir

Reviewer 3: Pourquoi mettre un lien vers la présentation de Git selon ... Github ?

 ```suggestion
- Un client ``git`` `Linux <https://git-scm.com/>`_ ou `Windows <https://gitforwindows.org/>`_.

    git push

Vous avez peut-être remarqué que cela ressemble à un triangle, avec un
segment manquant :

- Vous récupérez depuis *upstream* (le dépôt commun public sur Github)
- Vous poussez sur *origin* (votre clone sur Github)

C'est le travail de quelqu'un d'autre d'ajouter le dernier segment,
de votre *origin* au *upstream* public, pour « boucler la boucle ». C'est le
rôle des personnes qui *fusionnent* les *pull requests* après les avoir relues.

Vous avez peut-être aussi remarqué que vous n'avez jamais commité sur une
branche de version (``3.7``, ``3.8``, etc.), seulement récupéré les
modifications à partir d'elles.

Toutes les traductions sont faites sur la dernière version.
Nous ne traduisons jamais sur une version plus ancienne. Par exemple,
si la dernière version de python est Python 3.8, nous ne voulons pas
traduire directement sur la version python 3.5.
Si nécessaire, les traductions seraient rétroportées sur les versions
les plus anciennes par l'`équipe de documentation
<https://www.python.org/dev/peps/pep-8015/#documentation-team>`_.

Que traduire ?
~~~~~~~~~~~~~~

Vous pouvez utiliser `potodo`_, un outil fait pour trouver des fichiers ``po``
à traduire.

> Une fois installé, utiliser la commande ``potodo`` dans votre clone local.

Reviewer 3: ```suggestion
Une fois installé, utilisez la commande ``potodo`` dans votre clone local.
```

Vous pouvez choisir n'importe quel fichier non réservé dans la liste
renvoyée par la commande.

**Nous vous recommandons de ne pas commencer avec un fichier de ``c-api``
car c'est une partie très technique.**

Vous pouvez commencer par des tâches faciles comme réviser les entrées
*fuzzy* pour aider à garder la documentation à jour (trouvez les entrées
*fuzzy* l'aide de `make fuzzy`).

Vous pouvez également relire des entrées déjà traduites pour vous faire une

> idée,et passer ensuite à la traduction de celles qui ne sont pas traduites.

Reviewer 3: ```suggestion
idée, et passer ensuite à la traduction de celles qui ne sont pas traduites.
```

Vous pouvez les trouver à l'aide de `make todo`...

> Si vous traduisez un titre de lien, veuillez traduire le lien aussi.

Reviewer 3: ```suggestion
Vous pouvez aussi « traduire » des liens hypertextes
```

> (surtout si c'est un lien Wikipédia et que l'article a une traduction).

Reviewer 3: ```suggestion
(par exemple s'il s'agit d'un lien vers un article de Wikipédia qui possède une traduction).
Modifiez le lien et sa description dans ce cas.
```

| `Lien vers le repository <https://github.com/vpoulailleau/padpo>`__

Powrap
~~~~~~

| Formateur de fichier .po.
| Installez-le à l'aide de *pip* (``pip install powrap``) dans un
  environnement.
| `Lien vers le repository <https://github.com/JulienPalard/powrap>`__

Ressources de traduction
-------------------------

- le canal IRC `#python-fr <http://irc.lc/freenode/python-fr>`_ sur freenode ;
- la `liste traductions AFPy <http://lists.afpy.org/mailman/listinfo/traductions>`_ ;
- la `liste de diffusion doc-sig
  <https://mail.python.org/mailman/listinfo/doc-sig>`_ ;
- les `glossaires et dictionnaires de traduc.org
  <https://traduc.org/Glossaires_et_dictionnaires>`_, en particulier le
  `grand dictionnaire terminologique <http://gdt.oqlf.gouv.qc.ca/>`_
  de l'Office québécois de la langue française ;
- le `glossaire Python <https://docs.python.org/fr/3/glossary.html>`_, car
  il est déjà traduit ;
- le `guide stylistique pour le français de localisation des produits Sun
  <https://web.archive.org/web/20160821182818/http://frenchmozilla.org/FTP/TEMP/guide_stylistique_December05.pdf>`_ donne
  beaucoup de conseils pour éviter une traduction trop mot à mot ;

> - `deepl.com/translator <https://www.deepl.com>`_ ;

Reviewer 3: On peut résoudre #1030 ici :
```suggestion
- `deepl.com/translator <https://www.deepl.com>`_. Veuillez l'utiliser avec *extrême prudence*, deepl produit des traductions très « lourdes » à lire, avec des termes qui ne suivent pas les termes retenus dans le glossaire et truffées d'anglicismes. Les PRs trop inspirées de deepl seront **refusées**  ;
```

| `Lien vers le repository <https://github.com/vpoulailleau/padpo>`__

Powrap
~~~~~~

| Formateur de fichier .po.
| Installez-le à l'aide de *pip* (``pip install powrap``) dans un
  environnement.
| `Lien vers le repository <https://github.com/JulienPalard/powrap>`__

Ressources de traduction
-------------------------

- le canal IRC `#python-fr <http://irc.lc/freenode/python-fr>`_ sur freenode ;
- la `liste traductions AFPy <http://lists.afpy.org/mailman/listinfo/traductions>`_ ;
- la `liste de diffusion doc-sig
  <https://mail.python.org/mailman/listinfo/doc-sig>`_ ;
- les `glossaires et dictionnaires de traduc.org
  <https://traduc.org/Glossaires_et_dictionnaires>`_, en particulier le
  `grand dictionnaire terminologique <http://gdt.oqlf.gouv.qc.ca/>`_
  de l'Office québécois de la langue française ;
- le `glossaire Python <https://docs.python.org/fr/3/glossary.html>`_, car
  il est déjà traduit ;
- le `guide stylistique pour le français de localisation des produits Sun
  <https://web.archive.org/web/20160821182818/http://frenchmozilla.org/FTP/TEMP/guide_stylistique_December05.pdf>`_ donne
  beaucoup de conseils pour éviter une traduction trop mot à mot ;

> - `deepl.com/translator <https://www.deepl.com>`_. Veuillez l'utiliser avec *extrême prudence*, deepl produit des traductions très « lourdes » à lire, avec des termes qui ne suivent pas les termes retenus dans le glossaire et truffées d'anglicismes. Les PRs trop inspirées de deepl seront **refusées**  ;

Reviewer 3: 😃 
D'ailleurs je me demande si mettre « un esprit critique » ou « du recul » ne serait pas plus approprié.

Guide de contribution à la documention via GitHub.
===================================================

Prérequis
---------

- Un compte `Github <https://github.com/join>`_.
- Un client ``git`` `Linux <https://git-scm.com/>`_ ou `Windows <https://gitforwindows.org/>`_.
  https://gitforwindows.org/).

> - Un éditeur de fichier ``.po`` (comme `poedit <https://poedit.net/>`_).

Reviewer: préciser les que un éditeur de texte suffit mais pas recommandé ? et que des plugins existent pour emacs (https://www.gnu.org/software/gettext/manual/html_node/PO-Mode.html) et vim ?

Instructions :
--------------

Pour commencer vous aurez besoin de *forker* le dépôt des sources `python-docs-fr
<https://github.com/python/python-docs-fr>`_ en cliquant sur son bouton

> ``Fork``. Ceci crée une copie du projet sur votre compte Github : un endroit

Reviewer: ```suggestion
``Fork``. Ceci crée une copie du projet sur votre compte Github : un endroit
```

Reviewer: cc @awecx, les espaces insécables ;)

=> Si vous voyez :
| « -- » ou « --- » : faites :kbd:`Compose - - -`
| « ... » : faites :kbd:`Compose . . .`

Le cas de « "…" » :
~~~~~~~~~~~~~~~~~~~

Les guillemets français ``«`` et ``»`` ne sont pas identiques aux
guillemets anglais ``"``. Cependant, Python utilise les guillemets
anglais comme délimiteurs de chaîne de caractères. Il convient donc de
traduire les guillemets mais pas les délimiteurs de chaîne.

=> Si vous voyez :
| « "…" » : faites :kbd:`Compose < <` ou :kbd:`Compose > >`

Le cas de « :: »
~~~~~~~~~~~~~~~~

| Du point de vue du langage *reStructuredText* (ou *rst*) utilisé dans la
  documentation nous voyons soit « bla bla:: », soit « bla bla. :: ».
| ``::`` collé à la fin d'un mot signifie « affiche ``:`` et introduit un bloc de code »,
  mais un ``::`` après une espace signifie « introduit juste un bloc de code ».

En français, nous mettons une espace insécable devant nos deux-points, comme :
« Et voilà : ».

=> Traduisez ``mot deux-points deux-points`` par
``mot espace-insécable deux-points deux-points``.

Pour saisir une espace insécable faites : :kbd:`Compose SPACE SPACE`

Outils utiles pour la traduction
--------------------------------

> 

Reviewer: Peux tu rajouter poutils ? https://github.com/seluj78/poutils https://pypi.org/project/poutils/

Ressources de traduction
-------------------------

- les canaux IRC sur freenode :
  - `#python-docs-fr <http://irc.lc/freenode/python-docs-fr>`_
  Communauté python autour de la documentation française.
  - `#python-fr <http://irc.lc/freenode/python-fr>`_
  Communauté python française.
  - `#python-doc <http://irc.lc/freenode/python-fr>`_
  Communauté python autour de la documentation anglaise.
- la `liste traductions AFPy <http://lists.afpy.org/mailman/listinfo/traductions>`_ ;
- la `liste de diffusion doc-sig
  <https://mail.python.org/mailman/listinfo/doc-sig>`_ ;
- les `glossaires et dictionnaires de traduc.org
  <https://traduc.org/Glossaires_et_dictionnaires>`_, en particulier le
  `grand dictionnaire terminologique <http://gdt.oqlf.gouv.qc.ca/>`_
  de l'Office québécois de la langue française ;
- le `glossaire Python <https://docs.python.org/fr/3/glossary.html>`_, car
  il est déjà traduit ;
- le `guide stylistique pour le français de localisation des produits Sun
  <https://web.archive.org/web/20160821182818/http://frenchmozilla.org/FTP/TEMP/guide_stylistique_December05.pdf>`_ donne
  beaucoup de conseils pour éviter une traduction trop mot à mot ;

> - Wikipedia : En allant sur l'article d'un sujet sur la version anglaise

Reviewer 3: ```suggestion
- Wikipédia : En allant sur l'article d'un sujet sur la version anglaise
```

>   de Wikipedia, puis en basculant sur la version francaise pour voir

Reviewer 3: ```suggestion
  de Wikipédia, puis en basculant sur la version francaise pour voir
```

>   comment le sujet es traduit.

Reviewer 3: ```suggestion
  comment le sujet est traduit.
```
Donc on essaye de virer _toute_ référence à deepl ?
